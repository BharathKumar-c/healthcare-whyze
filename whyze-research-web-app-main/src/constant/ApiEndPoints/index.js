const FEASIBILITY_API = process.env.REACT_APP_FEASIBILITY_STUDY_BASE_URL;
const AUTH_API = process.env.REACT_APP_AUTH_URL;

export const feasibilityStudyApiUrls = {
  medicalConditionSearchUrl: `${FEASIBILITY_API}/search/indication`,
  medicationConditionSearchUrl: `${FEASIBILITY_API}/search/medication`,
  caseStudyUrl: `${FEASIBILITY_API}/case`,
  locationUrl: `${FEASIBILITY_API}/case/location`,
  initialPopulation: `${FEASIBILITY_API}/case/get-count`,
  allergiesListSearchUrl: `${FEASIBILITY_API}/search/allergies`,
  reactionListSearchUrl: `${FEASIBILITY_API}/search/reaction`,
  sitesDataUrl: `${FEASIBILITY_API}/case/site`,
  sitesWithPatientsCountUrl: `${FEASIBILITY_API}/case/site-patient-count`,
};

export const authEndPoints = {
  login: `${AUTH_API}/login`,
  register: `${AUTH_API}/register`,
  refreshToken: `${AUTH_API}/refreshToken`,
  forgotPassword: `${AUTH_API}/sendResetPasswordLink`,
  resetPassword: `${AUTH_API}/resetPassword`,
};

export const dashboardApiUrls = {
  projectUrl: `${FEASIBILITY_API}/project`,
  favouriteUrl: `${FEASIBILITY_API}/project/favourite`,
};

export const profileApiUrls = {
  image: `${AUTH_API}/image`,
  user: `${AUTH_API}/user`,
  profilePicture: `${AUTH_API}/profilePicture`,
};
