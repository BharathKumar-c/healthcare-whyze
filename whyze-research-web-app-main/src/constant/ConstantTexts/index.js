import React from 'react';

// dashboard welcome Screen const text

export const welComeComponentContent =
  'See how Whyze is preforming for researchers like you.';
export const createFirstStudy = 'Create your first study';
export const createFirstStudyContent =
  'Define your protocol or create your first feasibility study';
export const patientstitleLable = 'Patients';
export const totalPatientsLable = 'Total patients';
export const countriesLable = 'Countries';
export const sitesLable = 'Sites';
export const sitesRecruitedLable = 'Sites recruited';
export const avgTimeLable = 'Avg. response time';
export const trialsLable = 'Trials';
export const activeTrials = 'Active trials';
export const totalTrials = 'Total trials';
export const screenFailRate = 'Screen fail rate';
export const threeMonths = '3 months';
export const avgTimeToRecruit = 'Avg. time to recruit';
export const feasibilityStudiesInPlanning = 'Feasibility studies in planning';
export const seeAll = 'See all';

//Profile Page const text
export const ResearcherTitle = 'Researcher title -';
export const AuthProfile = 'AUTHENTICATED PROFILE';
export const ContactDetails = 'Contact details';
export const CompanyEmail = 'Company email address';
export const PhoneNumber = 'Phone';

//ContactModal Page Const texts

export const EmailAddress = 'Email address';
export const Phone = 'Phone';
export const okButtonLabel = 'Ok';
export const saveButtonLabel = 'save';
export const cancelButtonLabel = 'cancel';
export const contactDetailsLabel = 'Contact details';

//Addcriteria texts
export const addCriteriaTitle = 'Select a criteria item';

//RenameModal texts
export const renameModalTitle = 'Profile details';
export const fullNameLabel = 'Full name';
export const titleLabel = 'Title';

// Medication constant texts
export const medicationConstants = {
  medicationLabel: 'Medication',
  medicationErrorMessage: 'Please add the Medication',
  selectlabel: 'Select Dosage',
  inputLabel: 'Enter value',
  startedPrefixText: 'Started at least',
};
// Indication constant texts
export const indicationConstants = {
  medicalConditionLabel: 'Medical condition',
  medicalConditionErrorMessage: 'Please add the Medical condition',
};

//Allergies constant texte
export const allergiesConstants = {
  allergiesLabel: 'Allergies criteria',
  allergiesPlaceholder: 'Allergies',
  allergiesErrorText: 'Please add the Allergies',
};
//Symptom constant texts

export const symptomConstats = {
  symptomLabel: 'Symptom criteria',
  symptomPlaceholder: 'Symptom',
};
//Project settingmodal Constants

export const projectSettingConstants = {
  addProjectError: 'Please add the project',
  addProjectLabel: 'Project',
  addClientNameError: 'Please add ClientName',
  rangeLabel: 'Select',
  addClientLabel: 'Client name',
  description: 'Description',
  sharingLabel: 'Sharing',
  shareBtnLabel: 'Share',
  standardsText: 'Standards',
  standardDescription: 'Standardising markers and value outputs',
  measurementText: 'Default measurement',
  clinicalTrialError: 'Please add the clinical trial',
  clinicalTrial: 'Clinical Trial',
  sponsorError: 'Please add the sponsor',
  sponsor: 'Sponsor',
  projectSettings: 'Project settings',
  descriptionError: 'Please add the description',
  standardsError: 'Please select the standards',
  noIndication: 'No Indication',
};
export const shareProjectConstants = {
  addPeoplePlaceHolder: 'Add people and groups',
  accessPeopleText: 'People with access',
};
export const selectedPeoplesConstants = {
  addPeople: 'Add people and groups',
  notifyPeopleLabel: 'Notify people',
  messageLabel: 'Message',
};
// Nicotine constant texts
export const nicotineConstants = {
  nicotineConditionErrorMessage: 'Please add the nicotine condition',
  nicotineLabel: 'Nicotine criteria',
};

// Create Account constant texts
export const createAccountContants = {
  createAccounTitle: (
    <>
      Set up your <br /> Whyze account
    </>
  ),
  setUpWhyzeAccount: 'Set up your Whyze account',
  nameLabel: 'Name',
  emailLabel: 'Company email',
  phoneLabel: 'Your phone number',
  passwordLabel: 'Create password',
  characterLengthLabel: 'Between 8 and 20 characters',
  numericLabel: 'At least one number',
  uppercaseLabel: 'At least one upper-case letter',
  lowercaseLabel: 'At least one lower-case letter',
  createAccountButtonLabel: 'Create account',
  agreeConditionText: 'By clicking the button above you agree to our',
  termsOfUseText: 'Terms of use',
  privacyNoticesText: 'Privacy policies',
  alreadyHaveAccountText: 'Already have an account?',
  loginText: 'Login',
  termsOfUseLink: 'https://www.whyzehealth.com/privacy-notice/',
  privacyNoticesLink: 'https://www.whyzehealth.com/terms',
};

//indication const

export const conditions = {
  started: 'started',
  dosage: 'dosage',
  courseOfTherapy: 'course of therapy',
  diagnosed: 'diagnosed',
  recovered: 'recovered',
  severity: 'severity',
};

//demographics const

export const demographicsConst = {
  age: 'age',
  min: 'min',
  max: 'max',
  gender: 'gender',
};

//sites const

export const siteConst = {
  siteInformationCosnt: {
    patientSegmentHeader: 'Patient segment',
    connectedWithSite: 'Connected with this site',
    additionalCatchmentAreas:
      'Additional Whyze patients in site catchment areas',
    sitePerformance: 'Site performance',
    recruitmentTimeline: 'Recruitment timeline',
    dropOutRate: 'Drop out rate',
    trackRecordOfDoingTrials: 'Track record of doing trials',
    completedInThePastyear: 'Completed in the past year',
    primaryInvestigator: 'Primary Investigator',
    administration: 'Administration',
    screenFailRate: 'Screen fail rate',
    industryAverage: 'Industry average',
    avgResponseTime: 'Avg. response time',
  },
};

//setupAccount const

export const setupAccountConst = {
  dontHaveaccount: 'Don’t have an account yet?',
  setUpAccount: 'Set up account',
};

//forgotpassword const

export const forgotpasswordConst = {
  WHYZE: 'WHYZE',
  forgotPassword: 'Forgot Password',
  resendEmail: 'Enter your email below and we will send you a reset email.',
  pleaseTry: 'Didn’t receive it ? Please try again',
  returnTo: 'return to',
  loginPage: 'login',
};

//weight const  texts

export const weightConst = {
  weightRange: 'Weight Range',
  weightMin: 'min',
  weightMax: 'max',
  weightType: 'number',
  weightmaxErrorText: 'add max weight',
  weightminErrorText: 'add min weight',
  weightFrom: 'From',
  weightTo: 'To',
};

export const SaveAndCloseModalConst = {
  SaveAndCloseModalTitle: 'Name before exiting',
  projectNameInputLabel: 'Project',
  projectNameInputPlaceholder: 'New patient segment',
  clientNameInputLabel: 'Client name',
  descriptionNameInputLabel: 'Description',
  patientSegmentInputLabel: 'Patient segment required',
};

// Proposed Site list constant texts

export const proposedSiteListConstants = {
  title: 'Proposed Site List',
  seeMore: 'See more',
  emptyContent:
    'To create your proposed list, drag & drop or add your selected sites here.',
};

// Surgery constant text
export const surgeryConstants = {
  surgeryInputLabel: 'Surgery criteria',
  surgeryInputErrorMessage: 'Please add the Surgery',
};

export const feasibilityStudy = {
  newFeasibilityStudy: 'new-feasibility-study',
};
