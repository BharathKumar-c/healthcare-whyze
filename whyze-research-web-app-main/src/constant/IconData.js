import React from 'react';

import {
  Alcohol,
  Allergies,
  Availability,
  Bmi,
  Indication,
  Medication,
  Previousstudy,
  Surgery,
  Symptom,
  Weight,
  Nicotine,
  Demographics,
  LocationIcon,
} from '../assets';

export const IconList = [
  {
    Id: 1,
    Title: 'Indication',
    Icons: <Indication />,
  },
  { Id: 2, Title: 'Medication', Icons: <Medication /> },
  {
    Id: 3,
    Title: 'Surgery',
    Icons: <Surgery />,
  },
  {
    Id: 4,

    Title: 'Symptom',
    Icons: <Symptom />,
  },
  {
    Id: 5,
    Title: 'Alcohol',
    Icons: <Alcohol />,
  },
  {
    Id: 6,
    Title: 'Allergies',
    Icons: <Allergies />,
  },
  {
    Id: 7,
    Title: 'BMI',
    Icons: <Bmi />,
  },
  {
    Id: 8,
    Title: 'Nicotine',
    Icons: <Nicotine />,
  },
  {
    Id: 9,
    Title: 'Weight',
    Icons: <Weight />,
  },

  {
    Id: 10,

    Title: 'Availability',
    Icons: <Availability />,
  },
  {
    Id: 11,

    Title: 'Previous study',
    Icons: <Previousstudy />,
  },
  {
    Id: 12,
    Title: 'Demographics',
    Icons: <Demographics />,
  },
  {
    Id: 13,
    Title: 'Location',
    Icons: <LocationIcon />,
  },
];
