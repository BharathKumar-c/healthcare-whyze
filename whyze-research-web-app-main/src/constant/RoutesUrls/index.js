export const RoutesUrls = {
  dashBoardUrl: 'dashboard',
  newFeasibilityStudyUrl: 'new-feasibility-study/:id',
  notFoundUrl: '*',
  profileUrl: 'profile',
  loginUrl: '/',
  createAccountUrl: 'create-account',
  forgotPasswordUrl: 'forgot-password',
  setupPasswordUrl: 'setup-password',
  forgotPasswordEmailUrl: 'email-password',
  feasibilityStudiesInPlanning: 'feasibility-studies-in-planning',
};
