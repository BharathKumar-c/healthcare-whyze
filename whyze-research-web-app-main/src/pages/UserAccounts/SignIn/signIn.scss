@import '../../../common_styles/common_variable';

.sign-in-component {
    display: flex;
    align-items: center;
    flex-direction: column;
    height: 100vh;
    background-image: url('../../../assets/EllipseTop.svg'), url('../../../assets/EllipseBottom.svg');
    background-position: left top, right bottom;
    background-repeat: no-repeat, no-repeat;
    background-size: 300px 300px;
    padding: 2.5rem;

    &-logo {
        display: flex;
        justify-content: space-around;
        align-items: center;
        padding: 1rem;
        font-size: 1rem;
        font-weight: 700;
    }

    &-title {
        font-size: 36px;
        font-weight: 700;
        padding-top: 0.5rem;
        padding-bottom: 1rem;
    }

    &-input-box {
        display: flex;
        justify-content: space-around;
        flex-direction: column;
        padding: 0.5rem;
        width: 350px;
    }

    &-setupaccount-link {
        padding: 1rem;

        a {
            text-decoration: none;
            color: $anchor-tag;
        }
    }

    &-button-group {
        text-align: center;
        justify-content: center;
        display: flex;
        flex-direction: row;
        margin: 1.5rem;

        &_label{
            text-transform: none;
        }

        button {
            border-radius: 100px;
            font-size: 15px;
            font-weight: 500;
            text-transform: capitalize;
            padding: 0rem 8rem 0rem 8rem;
            height: 42px;
            width: 330px;


            &:nth-child(1) {
                background: $primary-color;
                color: white;
                border: none;

            }
        }
    }

    &-setup-acc {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        align-items: center;
        font-weight: 400;
        font-size: 16px;

        &-underline {
            margin-left: 0.5rem;
            text-decoration: underline;
            cursor: pointer;

            a {
                color: $anchor-tag;
            }
        }
    }
}