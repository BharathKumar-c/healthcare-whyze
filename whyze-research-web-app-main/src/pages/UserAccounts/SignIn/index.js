import React, { useState } from 'react';
import { HidePass, ShowpassIcon, WhyzeTextLogo } from '../../../assets';
import { useNavigate, useLocation } from 'react-router-dom';
import { Button, Input } from '../../../components/basic';
import './signIn.scss';
import ApiUtil from '../../../utils/ApiUtils';
import { authEndPoints } from '../../../constant/ApiEndPoints';
import {
  errorConstants,
  labelAndTitleConstants,
  regexConstants,
} from '../../../constant';
import { RoutesUrls } from '../../../constant/RoutesUrls';
import { message } from 'antd';

const SignIn = () => {
  const location = useLocation();
  const navigate = useNavigate();
  const [passwordType, setPasswordType] = useState(false);
  const [email, setEmail] = useState('');
  const [isEmailError, setIsEmailError] = useState(false);
  const [emailErrorText, setEmailErrorText] = useState('');
  const [password, setPassword] = useState('');
  const [isPasswordError, setIsPasswordError] = useState(false);
  const [passwordErrorText, setPasswordErrorText] = useState('');
  const [isLoading, setIsLoading] = useState(false);
  const [displayed, setDispalyed] = useState(false);

  const handleChange = (e) => {
    const name = e.target.name;
    const value = e.target.value;
    if (name === 'email') {
      setEmail(value);
      setIsEmailError(false);
      setEmailErrorText('');
    } else if (name === 'password') {
      setPassword(value);
      setIsPasswordError(false);
      setPasswordErrorText('');
    }
  };

  const handleSignIn = async () => {
    if (email === '' && password === '') {
      setIsEmailError(true);
      setIsPasswordError(true);
      setEmailErrorText(errorConstants.emailMustBeEntered);
      setPasswordErrorText(errorConstants.passwordMustBeEntered);
    } else if (email === '') {
      setIsEmailError(true);
      setEmailErrorText(errorConstants.emailMustBeEntered);
    } else if (password === '') {
      setIsPasswordError(true);
      setPasswordErrorText(errorConstants.passwordMustBeEntered);
    } else {
      const regexEmail = regexConstants.emailRegex.test(email);
      if (!regexEmail) {
        setIsEmailError(true);
        setEmailErrorText(errorConstants.enterValidEmailAddress);
      } else {
        setIsEmailError(false);
        setIsPasswordError(false);
        setEmailErrorText('');
        setPasswordErrorText('');
        setIsLoading(true);
        try {
          await ApiUtil.loginUser(authEndPoints.login, {
            email: email,
            password: password,
          }).then(() => {
            setIsLoading(false);
            navigate(RoutesUrls.dashBoardUrl);
          });
        } catch (error) {
          setIsLoading(false);
          console.log(error);
          setIsEmailError(true);
          setIsPasswordError(true);
          setPasswordErrorText(errorConstants.inCorrectEmailPassword);
        }
      }
    }
  };

  const displayMessage = (val) => {
    setDispalyed(true);
    if (location?.state?.from === 'forgotPassword') {
      return message.success(val);
    } else {
      return message.error(val);
    }
  };

  const handleKeyDown = (event) => {
    if (event.key === 'Enter') {
      handleSignIn();
    }
  };
  return (
    <div className="sign-in-component">
      {location?.state?.from === 'forgotPassword' && !displayed && (
        <div>{displayMessage(location?.state?.message)}</div>
      )}
      {location?.state?.from === 'resetPassword' && !displayed && (
        <div>{displayMessage(location?.state?.message)}</div>
      )}
      <div className="sign-in-component-logo">
        <WhyzeTextLogo />
      </div>
      <div className="sign-in-component-title">
        {labelAndTitleConstants.signIn}
      </div>
      <div className="sign-in-component-input-box">
        <Input
          id="main_textbox"
          lable="Company email"
          name={'email'}
          value={email}
          handleInputChange={(e) => handleChange(e)}
          errorText={emailErrorText}
          isError={isEmailError}
          onKeyDown={handleKeyDown}
        />
      </div>
      <div className="sign-in-component-input-box">
        <Input
          lable={'Enter password'}
          type={!passwordType ? 'password' : 'text'}
          name={'password'}
          value={password}
          suffix={
            !passwordType ? (
              <ShowpassIcon onClick={() => setPasswordType(!passwordType)} />
            ) : (
              <HidePass onClick={() => setPasswordType(!passwordType)} />
            )
          }
          handleInputChange={(e) => handleChange(e)}
          isError={isPasswordError}
          errorText={passwordErrorText}
          onKeyDown={handleKeyDown}
        />
      </div>
      <div className="sign-in-component-button-group">
        <Button
          handleButtonClick={() => handleSignIn()}
          label={labelAndTitleConstants.signIn}
          size={'large'}
          isLoading={isLoading}
        />
      </div>
      <div>
        {labelAndTitleConstants.forgotPassword}
        <a
          className="sign-in-component-setup-acc-underline"
          onClick={() => navigate('/forgot-password')}
        >
          {labelAndTitleConstants.clickHere}
        </a>
      </div>
    </div>
  );
};

export default SignIn;
