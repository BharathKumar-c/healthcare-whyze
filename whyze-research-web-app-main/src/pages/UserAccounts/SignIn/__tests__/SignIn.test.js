import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import SignIn from '..';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import ApiUtil from '../../../../utils/ApiUtils';
import { RoutesUrls } from '../../../../constant/RoutesUrls';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

jest.mock('../../../../utils/ApiUtils');

describe('Test cases for sign in', () => {
  test('Create the snapshots for signIn', () => {
    const { container } = render(<SignIn />);
    expect(container).toMatchSnapshot();
  });

  test('Whyze logo renders correctly', () => {
    const { container } = render(<SignIn />);
    const signInTitleText = container.getElementsByClassName(
      'sign-in-component-logo',
    );
    expect(signInTitleText[0].firstChild.textContent).toEqual('WhyzeLogo.svg');
    expect(signInTitleText[0].lastChild.textContent).toEqual('WHYZE');
  });

  test('Sign in title renders correctly', () => {
    const { container } = render(<SignIn />);
    const userTitleText = container.getElementsByClassName(
      'sign-in-component-title',
    );
    expect(userTitleText[0].textContent).toEqual('Sign in');
  });

  test('Check the text input count', () => {
    render(<SignIn />);
    const emailInputFields = screen.getAllByRole('textbox');
    const emailInputField = screen.getByRole('textbox');
    expect(emailInputFields).toHaveLength(1);
    expect(emailInputFields[0]).toBeInTheDocument();
    expect(emailInputField).toBeInTheDocument();
  });

  test('Should password input field renders', () => {
    const { container } = render(<SignIn />);
    const passwordInputFields = container.querySelector(
      'input[name="password"]',
    );
    expect(passwordInputFields).toBeInTheDocument();
  });

  test('Should render the Eye icon in password field', () => {
    const { container } = render(<SignIn />);

    const passwordInputFields = container.querySelector(
      'input[name="password"]',
    );

    expect(passwordInputFields.value).toMatch('');
    userEvent.type(passwordInputFields, 'test12345');
    expect(passwordInputFields.value).toBe('test12345');
  });

  test('Check the button', () => {
    render(<SignIn />);
    const userTitleText = screen.getAllByRole('button', {
      lable: 'Sign in',
    });
    expect(userTitleText).toHaveLength(1);
    expect(userTitleText[0]).toBeInTheDocument();
  });

  test('forget password text renders successFully!', () => {
    const { container } = render(<SignIn />);
    const forgetPasswordText = container.getElementsByClassName(
      'sign-in-component-setup-acc-underline',
    );
    expect(forgetPasswordText[0].textContent).toEqual('Forgot Password');
  });

  test('Set account text renders successFully!', () => {
    render(<SignIn />);
    const setUpAccountText = screen.getByRole('heading', {}, { level: 3 });
    expect(setUpAccountText).toBeInTheDocument();
  });

  test('Signin page to forgot-passowrd navigation', () => {
    render(
      <BrowserRouter>
        <SignIn />
      </BrowserRouter>,
    );
    expect(screen.getByText('Forgot Password')).toBeInTheDocument();
    userEvent.click(screen.getByText(/Forgot Password/i));
    expect(mockedUsedNavigate).toHaveBeenCalledWith('/forgot-password');
  });

  test('Signin page to setUp-account navigation', () => {
    render(
      <BrowserRouter>
        <SignIn />
      </BrowserRouter>,
    );
    expect(screen.getByText('Set up account')).toBeInTheDocument();
    userEvent.click(screen.getByText(/Set up account/i));
    expect(mockedUsedNavigate).toHaveBeenCalledWith(
      RoutesUrls.createAccountUrl,
    );
  });

  test('Should eye icon cirrectly', () => {
    const { container } = render(<SignIn />);
    screen.debug();
    const showpassIcon = container.getElementsByClassName('ant-input-suffix');

    expect(showpassIcon[0]).toBeInTheDocument();
    expect(showpassIcon[0].textContent).toEqual('ShowpassIcon.svg');
    fireEvent.click(showpassIcon[0].firstChild);
    expect(showpassIcon[0].textContent).toEqual('HidePass.svg');
    fireEvent.click(showpassIcon[0].firstChild);
    expect(showpassIcon[0].textContent).toEqual('ShowpassIcon.svg');
  });

  test('User sign in with empty email and passowrd field', async () => {
    const { container, getByRole, getByText } = render(
      <BrowserRouter>
        <SignIn />
      </BrowserRouter>,
    );
    const emailInput = container.querySelector('input[name="email"]');
    const passwordInputFields = container.querySelector(
      'input[name="password"]',
    );
    const submitButton = getByRole('button', { name: 'Sign in' });
    fireEvent.change(emailInput, '');
    fireEvent.change(passwordInputFields, '');
    userEvent.click(submitButton);
    const emailErrorField = getByText('Email address must be entered');
    const passwordErrorField = getByText('Password must be entered');
    expect(emailErrorField).toBeInTheDocument();
    expect(passwordErrorField).toBeInTheDocument();
  });
  test('User sign in with empty email field', async () => {
    const { container, getByRole, getByText } = render(
      <BrowserRouter>
        <SignIn />
      </BrowserRouter>,
    );
    const emailInput = container.querySelector('input[name="email"]');
    const passwordInputFields = container.querySelector(
      'input[name="password"]',
    );
    const submitButton = getByRole('button', { name: 'Sign in' });
    fireEvent.change(emailInput, '');
    userEvent.type(passwordInputFields, 'test12345');
    userEvent.click(submitButton);
    const emailErrorField = getByText('Email address must be entered');
    expect(emailErrorField).toBeInTheDocument();
  });
  test('User sign in with empty password field', async () => {
    const { container, getByRole, getByText } = render(
      <BrowserRouter>
        <SignIn />
      </BrowserRouter>,
    );
    const emailInput = container.querySelector('input[name="email"]');
    const passwordInputFields = container.querySelector(
      'input[name="password"]',
    );
    const submitButton = getByRole('button', { name: 'Sign in' });
    userEvent.type(emailInput, 'Test@testmail.com');
    fireEvent.change(passwordInputFields, '');
    userEvent.click(submitButton);
    const passwordErrorField = getByText('Password must be entered');
    expect(passwordErrorField).toBeInTheDocument();
  });
  test('User sign in with invalid email', async () => {
    const { container, getByRole, getByText } = render(
      <BrowserRouter>
        <SignIn />
      </BrowserRouter>,
    );
    const emailInput = container.querySelector('input[name="email"]');
    const passwordInputFields = container.querySelector(
      'input[name="password"]',
    );
    const submitButton = getByRole('button', { name: 'Sign in' });
    userEvent.type(emailInput, 'Testtestmail.com');
    userEvent.type(passwordInputFields, 'test12345');
    userEvent.click(submitButton);
    const emailErrorField = getByText('Enter valid email address');
    expect(emailErrorField).toBeInTheDocument();
  });
  test('User sign in', async () => {
    const { container, getByRole } = render(
      <BrowserRouter>
        <SignIn />
      </BrowserRouter>,
    );
    ApiUtil.loginUser.mockResolvedValueOnce({
      access_token:
        'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJ3aHl6ZSIsImVtYWlsIjoia2FydGhpa0BpdGVyby5pZSIsInVzZXJJRCI6IjAwMDc5Yzg2LTc1MjktNDY4NC1iOGNlLTQ1NDM0NjY0NWFiOCIsImlhdCI6MTY3NTc0OTg5NywiZXhwIjoxNjc1NzUzNDk3fQ.IcA7_XbweQFScijf9rdPDPn93uv4Uj6CMFVZILOfXmw',
      refresh_token:
        'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJ3aHl6ZSIsImVtYWlsIjoia2FydGhpa0BpdGVyby5pZSIsInVzZXJJRCI6IjAwMDc5Yzg2LTc1MjktNDY4NC1iOGNlLTQ1NDM0NjY0NWFiOCIsImlhdCI6MTY3NTc0OTg5NywiZXhwIjoxNjc1ODM2Mjk3fQ.B83W0J4LvEUAw5QtXOE_6XZ3AVh0L3LuWAqhsf92wLc',
    });
    const emailInput = container.querySelector('input[name="email"]');
    const passwordInputFields = container.querySelector(
      'input[name="password"]',
    );
    const submitButton = getByRole('button', { name: 'Sign in' });
    userEvent.type(emailInput, 'Test@testmail.com');
    userEvent.type(passwordInputFields, 'Test12345');
    userEvent.click(submitButton);

    await waitFor(() => {
      expect(ApiUtil.loginUser).toHaveBeenCalledWith('login', {
        email: 'Test@testmail.com',
        password: 'Test12345',
      });
      expect(ApiUtil.loginUser).toHaveBeenCalledTimes(1);
      expect(mockedUsedNavigate).toHaveBeenCalledWith(RoutesUrls.dashBoardUrl);
    });
  });
});
