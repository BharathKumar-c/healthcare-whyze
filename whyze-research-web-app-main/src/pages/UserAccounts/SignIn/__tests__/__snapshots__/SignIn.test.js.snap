// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for sign in Create the snapshots for signIn 1`] = `
<div>
  <div
    class="sign-in-component"
  >
    <div
      class="sign-in-component-logo"
    >
      <svg>
        WhyzeLogo.svg
      </svg>
      WHYZE
    </div>
    <div
      class="sign-in-component-title"
    >
      Sign in
    </div>
    <div
      class="sign-in-component-input-box"
    >
      <div
        id="main_textbox"
      >
        <span
          class="ant-typography input-lable"
        >
          Company email
        </span>
        <input
          class="ant-input"
          data-testid="main_textbox"
          name="email"
          placeholder=""
          style="border-radius: 0px;"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="sign-in-component-input-box"
    >
      <div>
        <span
          class="ant-typography input-lable"
        >
          Enter password
        </span>
        <span
          class="ant-input-affix-wrapper"
          style="border-radius: 0px;"
        >
          <input
            class="ant-input"
            name="password"
            placeholder=""
            type="password"
            value=""
          />
          <span
            class="ant-input-suffix"
          >
            <svg>
              ShowpassIcon.svg
            </svg>
          </span>
        </span>
      </div>
    </div>
    <div
      class="sign-in-component-button-group"
    >
      <button
        class="ant-btn ant-btn-default ant-btn-lg"
        type="button"
      >
        <span>
          Sign in
        </span>
      </button>
    </div>
    <div
      class="sign-in-component-setup-acc-underline"
    >
      <a>
        Forgot Password
      </a>
    </div>
    <div
      class="sign-in-component-setupaccount-link"
    >
      <h3>
        Don’t have an account yet?
         
        <a>
          Set up account
        </a>
      </h3>
    </div>
  </div>
</div>
`;
